### PR TITLE
Support pagination for submissions

### PR DESCRIPTION
**Pull request by andresuribe87**
# Overview
This is the first of many PRs that will address https://github.com/TBD54566975/ssi-service/issues/349. It focuses on supporting pagination for submissions, by building off of #518 .

# Description
This follows the same pattern as the existing one. Shared functionality was refactored into separate packages. 

# How Has This Been Tested?
Unit tests were implemented.

### CODE REVIEW

File: pkg/server/pagination/pagination.go [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+}
+
+func PageTokenQuery(c *gin.Context) url.Values {
+	query := c.Request.URL.Query()
+	delete(query, PageTokenParam)
+	delete(query, PageSizeParam)
+	return query
+}
+
+func MaybeSetNextPageToken(c *gin.Context, serviceNextPageToken string, respNextPageToken *string) bool {
```
Comment (decentralgabe) [inline pkg/server/pagination/pagination.go:78]: would help to add a comment on what this does, esp. why one param is nil-able

File: pkg/server/pagination/pagination.go [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+		}
+		encodedToken := base64.RawURLEncoding.EncodeToString(nextPageTokenData)
+		*respNextPageToken = encodedToken
+	}
+	return false
+}
+
+type PageRequest struct {
+	// When nil, all DIDs will be returned.
+	PageSize  *int    `json:"pageSize,omitempty"`
```
Comment (decentralgabe) [inline pkg/server/pagination/pagination.go:98]: is a page size of 0 valid?

File: pkg/server/pagination/pagination.go [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+	PageToken *string `json:"pageToken,omitempty"`
```
Comment (decentralgabe) [inline pkg/server/pagination/pagination.go:99]: similarly, empty string?

don't think these need to be ptrs

File: pkg/server/pagination/pagination.go [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+}
+
+func (r *PageRequest) ToServicePage() common.Page {
```
Comment (decentralgabe) [inline pkg/server/pagination/pagination.go:102]: need to be exported?

**Review by decentralgabe — APPROVED** [review]

File: pkg/server/pagination/pagination.go (hunk 4 above)
Comment (andresuribe87) [reply pkg/server/pagination/pagination.go:102]: Yes, it's used by other packages.

File: pkg/server/pagination/pagination.go (hunk 3 above)
Comment (andresuribe87) [reply pkg/server/pagination/pagination.go:117]: Added documentation. It's meant to mimic what was parsed from the request query, so empty means it wasn't provided.

File: pkg/server/pagination/pagination.go (hunk 2 above)
Comment (andresuribe87) [reply pkg/server/pagination/pagination.go]: It's not. Added validation.

File: pkg/server/pagination/pagination.go (hunk 1 above)
Comment (andresuribe87) [reply pkg/server/pagination/pagination.go:92]: Added documentation

**Pull request merged**